More helpful Python version check in setup.py

imphil opened the pull request:
In setup.py we check if Python is sufficiently recent. For users running
the installation from Python 2 this can be a blocker to install (and
subsequently use) cocotb, we therefore need to invest into making it
easy for users to diagnose and fix the problem. In many cases Python 3
is available on a user's system, but not used, commonly due to the use
of 'pip' instead of 'pip3'. Explicitly add a hint in this regard and
point to the documentation for more details.

To make the error message even more friendly, remove the backtrace
generated as part of `RuntimeError()` and use normal stderr instead. To
be able to support older Python 2 versions we don't use `print` but
`sys.stderr.write()`.

On Ubuntu 18.04/2.7.17:

Before:

```
philipp@philipp-VirtualBox:~/src/cocotb$ pip install .
Processing /home/philipp/src/cocotb
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-kZcdoF-build/setup.py", line 33, in <module>
        raise RuntimeError("Python version >= 3.5 required.")
    RuntimeError: Python version >= 3.5 required.

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-kZcdoF-build/
```

after:

```
philipp@philipp-VirtualBox:~/src/cocotb$ pip install .
Processing /home/philipp/src/cocotb
    Complete output from command python setup.py egg_info:
    This version of cocotb requires at least Python 3.5 to run.
    If you have Python 3 installed on your machine try
    using 'pip3' instead of 'pip' to install cocotb.
    For more information please refer to the documentation at
    https://cocotb.readthedocs.io.

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-bd_QP1-build/
```


<!--

Thanks for improving cocotb! Here are some points to make this as smooth as possible.
Not all of them may be applicable.

Most important: please explain *why* you are proposing this change.

* Make sure you have read https://github.com/cocotb/cocotb/blob/master/CONTRIBUTING.md
* Extend or add a test under `tests/test_cases/`.
* Add documentation under `documentation/source/`,
  docstrings in Python code, or Doxygen markup in C/C++ code.
  Use ``versionadded``/``versionchanged``/``deprecated``.
* Add a newsfragment - see `documentation/source/newsfragments/README.rst`.
* Use `closes #XXXX` to auto-close the issue that this PR fixes (if such).

-->

themperek commented on setup.py:
```diff
@@ -30,7 +30,20 @@
 import sys
 if sys.version_info[:2] < (3, 5):
-    raise RuntimeError("Python version >= 3.5 required.")
+    msg = ["This version of cocotb requires at least Python 3.5 to run."]
```
One could display what is the version installed.

eric-wieser commented on setup.py:
```diff
@@ -30,7 +30,20 @@
-    raise RuntimeError("Python version >= 3.5 required.")
+    msg = ["This version of cocotb requires at least Python 3.5 to run."]
+    if sys.version_info[0] == 2:
+        msg += [
+            "If you have Python 3 installed on your machine try ",
+            "using 'pip3' instead of 'pip' to install cocotb."
+        ]
```
Perhaps best here to recommend `python3 -m pip`, if `pip` was pointing at the wrong thing

eric-wieser commented on setup.py:
```diff
@@ -30,7 +30,20 @@
+        "https://cocotb.readthedocs.io."
+    ]
+
+    # The print() function is not available in older Python versions.
+    sys.stderr.write('\n'.join(msg))
+    sys.exit(1)
```
```suggestion
    raise SystemExit('\n'.join(msg))
```

imphil commented:
Addressed all review comments.

Note: I didn't use `sys.version` as it gives a rather long string, e.g. 


```
Python 2.7.17 (default, Nov  7 2019, 10:07:09)
    [GCC 7.4.0].
```
